featurex/extractors/image.py

```diff
@@ -63,7 +64,35 @@ class VibranceExtractor(ImageExtractor):
     def __init__(self):
         super(self.__class__, self).__init__()
 
-    def apply(self, img):
-        data = img.data
+    def apply(self, stim):
+        data = stim.data
         avg_color = np.var(data, 2).mean()
-        return Value(img, self, {'avg_color': avg_color})
+        return Value(stim, self, {'avg_color': avg_color})
+
+class SaliencyExtractor(ImageExtractor):
+
+    ''' Determines the saliency of the image using Itti & Koch (1998) algorithm implemented in pySaliencyMap '''
+
+    def __init__(self):
+        super(self.__class__, self).__init__()
+
+    def apply(self, stim):
+        # pySaliencyMap from https://github.com/akisato-/pySaliencyMap
+        data = stim.data
+
+        # Initialize variables
+        h, w, c = stim.data.shape
+        sm = pySaliencyMap.pySaliencyMap(h, w)
+
+        # Compute saliency maps and store full maps as derivatives
+        stim.derivatives = dict()
```

Comment:
Just making a note here: at some point, we'll probably want to add an economy flag to the package so that intermediate or derivative files aren't kept around if the user doesn't want them. We would then want to check against that flag here and conditionally store these derivatives.